crates/amaru-ledger/src/state.rs

```diff
@@ -541,6 +541,26 @@ impl<S: Store, HS: HistoricalStores> State<S, HS> {
         Ok(result)
     }
 
+    /// Calcualte chain density over the last `k` blocks (or oldest block in the volatileDB) given some `Point`.
+    /// If the `Point` is older than the oldest block in the volatileDB, density is 0
+    pub fn chain_density(&self, point: &Point) -> f64 {
+        let latest_slot = point.slot_or_default();
+        let k_slot = self
+            .volatile
+            .view_front()
+            .map(|state| &state.anchor.0)
+            .unwrap_or(&Point::Origin)
+            .slot_or_default();
+
+        if k_slot > latest_slot {
+            0f64
+        } else {
+            // Add one to the number of blocks in the volatileDB because we are including the `Point` in the chain density
+            (self.volatile.len() as f64 + 1f64)
```

Comment:
Note that apart from _just after the startup_ (literally within the second that follows), self.volatile.len() is always equal to `k`, by construction.

Reply:
Yeah, that shouldn't be an issue